### ISSUE TITLE
`<td onMouseDown={...} />` should trigger  jsx-a11y/no-noninteractive-element-interactions

### ISSUE THREAD

**Issue by k-funk** [jsx-eslint/eslint-plugin-jsx-a11y#1064]
I just upgraded from eslint 6 to 9, and in the process, discovered that there was a change to no longer warn on this

```
<div onMouseDown={() => console.log('mouse down')}>Mouse down</div>
```

Seems it was very intentional. Curious of the motivation?
https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/src/util/isNonInteractiveElement.js#L82

**Comment by k-funk** [comment]
My mistake, i'm actually doing eslint 8 -> 9.

I made a minirepo to show:
* eslint v9 + eslint-plugin-jsx-a11y v6.10: branch [master](https://github.com/k-funk/eslint-plugin-jsx-a11y-bug1)
* eslint v8 + eslint-plugin-jsx-a11y v6.9: branch [v6.9](https://github.com/k-funk/eslint-plugin-jsx-a11y-bug1/tree/v6.9)

https://github.com/user-attachments/assets/6fda44e7-a64e-43c7-9b37-cdfa81c4053d

**Comment by ljharb** [comment]
I would assume that if you upgrade this plugin *first* - before upgrading eslint - you'll see the same change? (generally it's best to update one thing at a time anyways)

Assuming that's the case, then it's likely due to an `aria-query` update (referenced by https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/commit/4925ba8d0bf80a4b1d8e8645d310590bf1b40b64) bringing the plugin more in line with proper role classifications on the web.
